### PR TITLE
{bio}[foss/2017b,intel/2017b] MSM v1.0

### PR DESCRIPTION
**Pull request by vanzod**
Requires ~~https://github.com/easybuilders/easybuild-easyblocks/pull/1770~~

### CODE REVIEW

**Comment by vanzod** [comment]
Test report by @vanzod
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
cermis - Linux debian 9.9, Intel(R) Xeon(R) CPU           X5670  @ 2.93GHz, Python 2.7.13
See https://gist.github.com/642dd19e0e9512495cd40dfbb0e89693 for a full test report.

**Comment by bartoldeman** [comment]
reopening for travis now the easyblock has been merged

**Comment by migueldiascosta** [comment]
Test report by @migueldiascosta
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
grc-cluster1 - Linux centos 6.10, Intel(R) Xeon(R) CPU E5-2640 0 @ 2.50GHz, Python 2.7.14
See https://gist.github.com/fbb35eb976c40cbd91b98352c5589bad for a full test report.

**Comment by migueldiascosta** [comment]
Test report by @migueldiascosta
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in this PR)
sms - Linux centos linux 7.6.1810, AMD EPYC 7601 32-Core Processor, Python 2.7.5
See https://gist.github.com/9087c5bd381c2cccb4e5d77b0a855953 for a full test report.

**Comment by migueldiascosta** [comment]
weird, I don't have a problem building other stuff that depends on `libpng` with `intel/2017b` on this system...

**Comment by boegel** [comment]
@migueldiascosta see https://github.com/easybuilders/easybuild-easyconfigs/issues/5033?

**Comment by migueldiascosta** [comment]
@boegel yes, adding `zlib` as a direct dependency solves it...

So, a direct dependency is needed althrought there's an indirect one, but only in the `intel` toolchain (and only on CentOS7?)... weird

(the system `zlib` on CentOS7 is 1.2.7...)

**Comment by vanzod** [comment]
@migueldiascosta Can you please test it again? Ideally after merging https://github.com/easybuilders/easybuild-easyblocks/pull/1775

**Comment by vanzod** [comment]
Test report by @vanzod
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
cermis - Linux debian 9.9, Intel(R) Xeon(R) CPU           X5670  @ 2.93GHz, Python 2.7.13
See https://gist.github.com/f11f58665299f043d8b6925ff55834a9 for a full test report.

File: easybuild/easyconfigs/m/MSM/MSM-1.0-intel-2017b.eb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    'estimate_metric_distortion', 'msm', 'msmapplywarp',
+    'msm_metric_sim', 'msmresample', 'surfconvert'
+]
+
+files_to_copy = [(['src/MSM/' + f for f in _binfiles], 'bin')]
+
+sanity_check_paths = {
+    'files': ['bin/' + f for f in _binfiles],
+    'dirs': ['doc', 'include', 'lib']
+}
```
Comment (boegel) [inline easybuild/easyconfigs/m/MSM/MSM-1.0-intel-2017b.eb]: @vanzod Why isn't the `files_to_copy` and `sanity_check_paths` part not handled by the custom easyblock for `MSM`?

File: easybuild/easyconfigs/m/MSM/MSM-1.0-intel-2017b.eb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+version = "1.0"
+
+homepage = "https://github.com/ecr05/MSM_HOCR"
+description = """Multimodal Surface Matching with Higher order Clique Reduction"""
+
+toolchain = {'name': 'intel', 'version': '2017b'}
+
+source_urls = ['https://github.com/ecr05/MSM_HOCR/archive']
+sources = ['%(version)s.tar.gz']
+patches = ['MSM-1.0_Fix_abs_overload.patch']
```
Comment (boegel) [inline easybuild/easyconfigs/m/MSM/MSM-1.0-intel-2017b.eb]: please use `%(version)s` rather than hardcoding the `1.0` here?

**Comment by vanzod** [comment]
The simplified easyconfig files require https://github.com/easybuilders/easybuild-easyblocks/pull/1776/files

**Comment by vanzod** [comment]
Test report by @vanzod
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
cermis - Linux debian 9.9, Intel(R) Xeon(R) CPU           X5670  @ 2.93GHz, Python 2.7.13
See https://gist.github.com/fc7375a98770737e87135d25b7038259 for a full test report.

**Comment by boegel** [comment]
@vanzod Ignore the failing test, that'll get fixed once the easyblock PR is merged, I'll re-trigger the tests here afterwards...

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2012.delcatty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/a59ee244cf950ea858c3b8c78ef73074 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2425.golett.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/202958523af0f899a018b469192199c3 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @vanzod!

**Pull request merged**